optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/PlannerBenchmarkRunner.java

```diff
@@ -122,54 +124,94 @@ private void initBenchmarkDirectoryAndSubdirs() {
     }
 
     private void warmUp() {
-        if (plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit() > 0L) {
-            logger.info("================================================================================");
-            logger.info("Warm up started");
-            logger.info("================================================================================");
-            long startingTimeMillis = System.currentTimeMillis();
-            long timeLeft = plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit();
-            List<ProblemBenchmarkResult> unifiedProblemBenchmarkResultList = plannerBenchmarkResult.getUnifiedProblemBenchmarkResultList();
-            Iterator<ProblemBenchmarkResult> it = unifiedProblemBenchmarkResultList.iterator();
-            while (timeLeft > 0L) {
-                if (!it.hasNext()) {
-                    it = unifiedProblemBenchmarkResultList.iterator();
-                }
-                ProblemBenchmarkResult problemBenchmarkResult = it.next();
-                timeLeft = warmUp(problemBenchmarkResult, startingTimeMillis, plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit(), timeLeft);
-            }
-            logger.info("================================================================================");
-            logger.info("Warm up ended");
-            logger.info("================================================================================");
+        if (plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit() <= 0L) {
+            return;
+        }
+        logger.info("================================================================================");
+        logger.info("Warm up started");
+        logger.info("================================================================================");
+        long timeLeft = plannerBenchmarkResult.getWarmUpTimeMillisSpentLimit();
+        long timeEnd = System.currentTimeMillis() + timeLeft;
+        Map<SingleBenchmarkRunner, Future<SingleBenchmarkRunner>> futureMap
+                = new HashMap<SingleBenchmarkRunner, Future<SingleBenchmarkRunner>>();
+        Map<SolverBenchmarkResult, TerminationConfig> originalTerminationConfigMap
+                = new HashMap<SolverBenchmarkResult, TerminationConfig>();
+        while (timeLeft > 0L) {
+            warmUpStarted(futureMap, originalTerminationConfigMap, timeLeft);
+            warmUp(futureMap, timeEnd);
+            timeLeft = timeEnd - System.currentTimeMillis();
+            futureMap.clear();
+        }
+        warmUpEnded(originalTerminationConfigMap);
+        warmUpExecutorService.shutdownNow(); // Terminate warm up solvers early
+        List<Runnable> notFinishedWarmUpList = warmUpExecutorService.shutdownNow(); // Make sure they're terminated
```

Comment:
They were needed to 1. kill the threads 2. verify their dead. Not needed anymore.